### PR TITLE
Fix UnicodeDecodeError on openssl generated signatures

### LINKED ISSUE #23: Retrieving signature serial number raises UnicodeDecodeError
From sentry: https://sentry.prod.mozaws.net/operations/olympia-dev/issues/616151/

```python
UnicodeDecodeError: 'charmap' codec can't decode byte 0x7b in position 0: character maps to <undefined>
  File "django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "newrelic/hooks/framework_django.py", line 527, in wrapper
    return wrapped(*args, **kwargs)
  File "olympia/amo/decorators.py", line 33, in wrapper
    return func(request, *args, **kw)
  File "olympia/editors/decorators.py", line 33, in wrapper
    return f(request, *args, **kw)
  File "olympia/addons/decorators.py", line 45, in wrapper
    return f(request, addon, *args, **kw)
  File "olympia/editors/views.py", line 674, in review
    form.helper.process()
  File "olympia/editors/helpers.py", line 516, in process
    return self.actions[action]['method']()
  File "olympia/editors/helpers.py", line 674, in process_public
    sign_file(file_, settings.SIGNING_SERVER)
  File "olympia/lib/crypto/packaged.py", line 142, in sign_file
    cert_serial_num = unicode(call_signing(file_obj, endpoint))
  File "olympia/lib/crypto/packaged.py", line 96, in call_signing
    cert_serial_num = get_signature_serial_number(pkcs7)
  File "signing_clients/apps.py", line 378, in get_signature_serial_number
    content = cms.ContentInfo.load(pkcs7)['content'].native
  File "asn1crypto/core.py", line 3706, in native
    raise e
```

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #23

### CODE REVIEW

File: signing_clients/apps.py [hunk 1]
```diff
@@ -15,12 +15,12 @@
 from six import PY3, string_types, text_type, python_2_unicode_compatible
 from six.moves import cStringIO as StringIO
 
-from asn1crypto import cms
+from asn1crypto import cms, core as asn1core
 
-# Lame hack to take advantage of a not well known OpenSSL flag.  This omits
-# the S/MIME capabilities when generating a PKCS#7 signature.  If included,
-# XPI signature verification breaks.
-PKCS7_NOSMIMECAP = 0x200
```
Comment (diox) [inline signing_clients/apps.py]: Why is this removed?

Comment (EnTeQuAk) [inline signing_clients/apps.py]: It's still from m2crypto ages and was unused 😇

**Comment by EnTeQuAk** [comment]
@jvehent can you give this a review please? I'm working with upstream on a proper fix but in the meantime this monkey-patch is needed to make this work for our certificates. Note all the comments and explanations in the related issues opened.

File: signing_clients/apps.py [hunk 2]
```diff
@@ -15,12 +15,12 @@
...
+# Patch asn1crypto teletex codec to actually be latin 1 (iso-8859-1)
+# See https://github.com/wbond/asn1crypto/issues/60 and
+# https://github.com/mozilla/signing-clients/issues/23 for more details
+asn1core.TeletexString._encoding = 'latin1'
```
Comment (jvehent) [inline signing_clients/apps.py]: This seem to have implications on the way we manage addons in general.
Would it effectively reject any addon with a non latin1 character in the ID?
Do we have any such addons today?
Why not use a UTF8String instead?

**Comment by EnTeQuAk** [comment]
It wouldn't reject them, this encoding is only used to read the serial
number and m2crypto (and thus openssl directly) use Latin1 to encode
those strings in trunion so we have to read them in latin1 too I think.

On Wed, Jun 28, 2017, at 09:13 PM, Julien Vehent [:ulfr] wrote:
> *@jvehent* commented on this pull request.


> 
> In signing_clients/apps.py[1]:


> > -# Lame hack to take advantage of a not well known OpenSSL flag.
> > This omits -# the S/MIME capabilities when generating a PKCS#7
> > signature.  If included, -# XPI signature verification breaks. -
> > PKCS7_NOSMIMECAP = 0x200 +# Patch asn1crypto teletex codec to
> > actually be latin 1 (iso-8859-1) +# See
> > https://github.com/wbond/asn1crypto/issues/60 and +#
> > https://github.com/mozilla/signing-clients/issues/23 for more
> > details +asn1core.TeletexString._encoding = 'latin1'
>> 


> This seem to have implications on the way we manage addons in general.> Would it effectively reject any addon with a non latin1 character
> in the ID?> Do we have any such addons today?
> Why not use a UTF8String instead?
> 


> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].> 

Links:

  1. https://github.com/mozilla/signing-clients/pull/24#discussion_r124551906
  2. https://github.com/mozilla/signing-clients/pull/24#pullrequestreview-46850976

**Comment by jvehent** [comment]
Cool. r+ from me then.
